src/module.cc

```diff
@@ -0,0 +1,388 @@
+  if (core_dump_switch != NULL) {
+    nodereport_core = ProcessNodeReportCoreSwitch(core_dump_switch);
+  }
+  const char* trigger_signal = secure_getenv("NODEREPORT_SIGNAL");
+  if (trigger_signal != NULL) {
+    nodereport_signal = ProcessNodeReportSignal(trigger_signal);
+  }
+  const char* report_name = secure_getenv("NODEREPORT_FILENAME");
+  if (report_name != NULL) {
+    ProcessNodeReportFileName(report_name);
+  }
+  const char* directory_name = secure_getenv("NODEREPORT_DIRECTORY");
+  if (directory_name != NULL) {
+    ProcessNodeReportDirectory(directory_name);
+  }
+
+  // If NodeReport requested for fatalerror, set up the V8 call-back
+  if (nodereport_events & NR_FATALERROR) {
+    isolate->SetFatalErrorHandler(OnFatalError);
+    error_hook_initialised = true;
+  }
+
+  // If NodeReport requested for exceptions, tell V8 to capture stack trace and set up the callback
+  if (nodereport_events & NR_EXCEPTION) {
+    isolate->SetCaptureStackTraceForUncaughtExceptions(true, 32, v8::StackTrace::kDetailed);
+    // The hook for uncaught exception won't get called unless the --abort_on_uncaught_exception option is set
+    v8::V8::SetFlagsFromString("--abort_on_uncaught_exception", sizeof("--abort_on_uncaught_exception")-1);
+    isolate->SetAbortOnUncaughtExceptionCallback(OnUncaughtException);
+    exception_hook_initialised = true;
+  }
+
+#ifndef _WIN32
+  // If NodeReport requested on external user signal set up watchdog thread and callbacks
+  if (nodereport_events & NR_SIGNAL) {
+    SetupSignalHandler();
+  }
+#endif
+
+  exports->Set(Nan::New("triggerReport").ToLocalChecked(),
+               Nan::New<v8::FunctionTemplate>(TriggerReport)->GetFunction());
+  exports->Set(Nan::New("setEvents").ToLocalChecked(),
+               Nan::New<v8::FunctionTemplate>(SetEvents)->GetFunction());
+  exports->Set(Nan::New("setCoreDump").ToLocalChecked(),
+               Nan::New<v8::FunctionTemplate>(SetCoreDump)->GetFunction());
+  exports->Set(Nan::New("setSignal").ToLocalChecked(),
+               Nan::New<v8::FunctionTemplate>(SetSignal)->GetFunction());
+  exports->Set(Nan::New("setFileName").ToLocalChecked(),
+               Nan::New<v8::FunctionTemplate>(SetFileName)->GetFunction());
+  exports->Set(Nan::New("setDirectory").ToLocalChecked(),
+               Nan::New<v8::FunctionTemplate>(SetDirectory)->GetFunction());
+  exports->Set(Nan::New("setVerbose").ToLocalChecked(),
+               Nan::New<v8::FunctionTemplate>(SetVerbose)->GetFunction());
+
+  if (nodereport_verbose) {
+#ifdef _WIN32
+    fprintf(stdout, "nodereport: initialization complete, event flags: %#x core flag: %#x\n",
+            nodereport_events, nodereport_core);
+#else
+    fprintf(stdout, "nodereport: initialization complete, event flags: %#x core flag: %#x signal flag: %#x\n",
+            nodereport_events, nodereport_core, nodereport_signal);
+#endif
+  }
+}
+
+NODE_MODULE(nodereport, Initialize)
+
+}  // namespace nodereport
+
```

Comment:
General observation: there is a somewhat inconsistent mix of nullptr vs. NULL in this file.  Since you're going for C++11, I'd use nullptr everywhere.